### PR TITLE
Adds ability to define description on posts for meta descriptions.

### PR DESCRIPTION
**Pull request by samjonester**
A new description field is added to the meta content at the top of a post's markdown file. This is used to generate the <meta name="description"/> tag.

Also adds descriptions to @samjonester 's posts as an example.

### CODE REVIEW

**Comment by searls** [comment]
LGTM, will merge.

Is it still the case that meta descriptions should be capped at 160 chars? If so, you may want to shorten your examples.

**Comment by jasonkarns** [comment]
I'm not sure if recent posts still have this, but we used to have a `tldr` field, which itself had a `title` and `body`. I suspect that the `tldr.title` field would be exactly what we want for most descriptions, and would like to see if it could be leveraged.

**Comment by searls** [comment]
@jason -- we haven't used it recently because I was never in love with the style of the tldr blocks after we redesigned the blog, but it's still there.

I thought similarly when I first read this PR, but I think too many of our tldr titles were more of cute subtitles and probably wouldn't match the intent well TBQH

**Comment by jasonkarns** [comment]
How about rendering the tldr.title field as a fallback when the meta description is empty? This would also cover the (probably common) case where the author doesn't _want_ a TLDR block rendered at all, but still needs the meta description.

File: app/posts/2017-01-18-life-after-nil.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "Life After Nil"
+description: 'Learn about the benifits that a type system can add to your coding by comparing Ruby to Haskell. We will “forget about nil”, “declaratively model your domain”, and “allow your compiler to drive your design”.'
```
Comment (jasonkarns) [inline app/posts/2017-01-18-life-after-nil.md]: were the smart quotes intentional?

**Comment by searls** [comment]
I'm :-1: on that proposal, Jason. Authors don't check metatags, and it's more likely IMO you'll get nonsensical meta tags than the tldr.title will actually improve the search description.

Here's a sample of tldr titles:

```
app/posts/2013-11-12-1st-class-web-development-with-lineman.md
10:  title: "Making front-end development more delightful."

app/posts/2013-08-29-great-technical-talks.md
7:  title: "Dramatic README recitations need not apply."

app/posts/2013-12-04-calls-for-transparency.md
7:  title: "An appeal to fairness"

app/posts/2014-04-19-setting-up-a-serious-ember-project.md
12:  title: "Introduction to Ember.js"

app/posts/2014-05-02-ember-routes-models-templates-and-controllers.md
12:  title: "Introduction to Ember.js"

app/posts/2013-12-06-angular-onramp.md
4:  title: "A place to use our new toys"

app/posts/2014-07-17-test-angular-like-it-was-made-out-of-javascript.md
10:  title: "Misdirection For Profit"

app/posts/2015-11-10-introducing-ember-cli-deploy-cloudfront-and-ember-cli-deploy-aws-pack.md
7:  title: "Two new plugins!"

app/posts/2015-11-13-concurrent-ruby-1.0.md
7:  title: "Concurrent Ruby 1.0"

app/posts/2016-02-01-react-reconciliation.md
7:  title: "Reconcilable Differences with React and VDOM"

app/posts/2017-01-05-react-d3.md
8:  title: "React and D3: Stronger Together"
```

File: app/posts/2017-01-18-life-after-nil.md (hunk 1 above)
Comment (samjonester) [inline app/posts/2017-01-18-life-after-nil.md]: Ugh Copy Paste.

**Comment by samjonester** [comment]
@searls, 2 things:
* I agree with you for the `tldr` situation. The `tldr` section will more likely have a negative effect.
* It looks like meta descriptions are indeed still capped at 160 chars, so I will need to update my examples. Thanks for the heads up.

**Pull request merged**